File: Algorithms/Sortings/Java/quick_sort_lomuto.java
import java.util.Arrays;

public class Lomuto {
	public static void main(String[] args) {
		int[] array = new int[] {0, 5, -2, 7, 3};
		qiuckSort(array);
		System.out.println(array);
	}

	public static void qiuckSort(int[] array) {
		qiuckSort(array, 0, array.length - 1);
	}

	public static void qiuckSort(int[] secuence, int lo_index, int hi_index) {
		if(lo_index >= hi_index) return;

		int pivot = partition(secuence, lo_index, hi_index);
		qiuckSort(secuence, lo_index, pivot - 1);
		qiuckSort(secuence, pivot + 1, hi_index);
	}

	public static int partition(int[] secuence, int lo_index, int hi_index) {
		int support_element = secuence[lo_index];
		int j = lo_index;
		for(int i = lo_index + 1; i <= hi_index; i++) {
			if(secuence[i] < support_element) {
				j += 1;
				swap(secuence, i, j);
			}
		}
		swap(secuence, lo_index, j);
		return j;
	}

	public static void swap(int[] arr, int i, int j) {
		int temp = arr[i];
		arr[i] = arr[j];
		arr[j] = temp;
	}
}

File: Algorithms/Sortings/Python/quick_sort_lomuto.py
def quick_sort(secuence, lo_index=None, hi_index=None):
	if lo_index is None:
		lo_index = 0
	if hi_index is None:
		hi_index = len(secuence) - 1
	if lo_index >= hi_index:
		return None
	pivot = partition(secuence, lo_index, hi_index)
	quick_sort(secuence, lo_index, pivot - 1)
	quick_sort(secuence, pivot + 1, hi_index)

def partition(secuence, lo_index, hi_index):
	support_element = secuence[lo_index]
	j = lo_index
	for i in range(lo_index + 1, hi_index + 1):
		if secuence[i] < support_element:
			j += 1
			secuence[j], secuence[i] = secuence[i], secuence[j]
	secuence[lo_index], secuence[j] = secuence[j], secuence[lo_index]
	return j


secuence = [0, 5, -2, 7, 3]
print(quick_sort(secuence))
print(secuence)